raco/rules.py

```diff
@@ -406,8 +406,11 @@ def descend_tree(op, cond):
     def fire(self, op):
         if not isinstance(op, algebra.Select):
             return op
-        if op.has_been_pushed:
-            return op
+        if hasattr(op, "has_been_pushed"):
+            if op.has_been_pushed:
+                return op
+            else:
+                 op.has_been_pushed = False
```

Comment:
This seems redundant. If has_been_pushed is False, make it False. Or is it that has_been_pushed is 0 or None somewhere?